packages/cashier-v2/src/lib/Transfer/hooks/useExtendedTransferAccounts.ts

```diff
@@ -0,0 +1,99 @@
+import { useActiveAccount, useCurrencyConfig } from '@deriv/api-v2';
+import { THooks } from '../../../hooks/types';
+
+type TGetCurrencyConfig = ReturnType<typeof useCurrencyConfig>['getConfig'];
+
+const sortedMT5Accounts = (accounts: THooks.TransferAccounts, getConfig: TGetCurrencyConfig) => {
+    return accounts
+        .filter(account => account.account_type === 'mt5')
+        .map(account => ({
+            ...account,
+            currencyConfig: account?.currency ? getConfig(account.currency) : undefined,
+        }));
+};
+
+const derivCTrader = (accounts: THooks.TransferAccounts, getConfig: TGetCurrencyConfig) => {
+    return accounts
+        .filter(account => account.account_type === 'ctrader')
+        .map(account => ({
+            ...account,
+            currencyConfig: account?.currency ? getConfig(account.currency) : undefined,
+        }));
+};
+
+const derivXAccount = (accounts: THooks.TransferAccounts, getConfig: TGetCurrencyConfig) =>
+    accounts
+        .filter(account => account.account_type === 'dxtrade')
+        .map(account => ({
+            ...account,
+            currencyConfig: account?.currency ? getConfig(account.currency) : undefined,
+        }));
+
+const fiatDerivAccounts = (accounts: THooks.TransferAccounts, getConfig: TGetCurrencyConfig) => {
+    return accounts
+        .filter(
+            account => account.account_type === 'binary' && account.currency && getConfig(account.currency)?.is_fiat
+        )
+        .map(account => ({
+            ...account,
+            currencyConfig: account?.currency ? getConfig(account.currency) : undefined,
+        }));
+};
+
+const sortedCryptoDerivAccounts = (accounts: THooks.TransferAccounts, getConfig: TGetCurrencyConfig) => {
+    return accounts
+        .filter(
+            account => account.account_type === 'binary' && account.currency && getConfig(account.currency)?.is_crypto
+        )
+        .map(account => ({
+            ...account,
+            currencyConfig: account?.currency ? getConfig(account.currency) : undefined,
+        }))
+        .sort((prev, next) => {
+            return prev.currency && next.currency ? prev.currency.localeCompare(next.currency) : 0;
+        });
+};
+
+/**
+    A hook which modifies the accounts received from `transfer_between_accounts` response as follows
+    - appends `currency_config` data for each account
+    - sorts the mt5 accounts based on group type
+    - sorts the crypto accounts alphabetically
+*/
+const useExtendedTransferBetweenAccounts = (accounts: THooks.TransferAccounts) => {
+    // console.log('=> hook - accounts', accounts);
+    const { data: activeAccount, isLoading: isActiveAccountLoading } = useActiveAccount();
+    const { getConfig, isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+
+    const isLoading = !accounts || isActiveAccountLoading || isCurrencyConfigLoading;
+
+    const extendedTransferableAccounts = !isLoading
```

Comment:
can we please memoize these calculated Non-primitive data values so they don't get reinitialized on render when we consume them